Assets/Scripts/Framework/Gameplay/LevelTracker.cs

```diff
@@ -0,0 +1,208 @@
+        /// An action that broadcasts when a package is delivered and the amount of points that is added.
+        /// </summary>
+        public Action<int> OnPackageDelivered;
+
+        private static readonly char[] CHARACTERS = "QWERTYUIOPASDFGHJKLZXCVBNM".ToCharArray();
+
+        public override void OnStartServer()
+        {
+            base.OnStartServer();
+
+            SelectLevelToLoad(0);
+        }
+
+        /// <summary>
+        /// A function that is used to load a new level.
+        /// </summary>
+        /// <param name="index">The level ID.</param>
+        [Server]
+        private void SelectLevelToLoad(int index)
+        {
+            _currentScore = 0;
+
+            StationLevelData[] levelData = _levels.levels[index].StationLevelData;
+            List<string> labels = GenerateLabelsForStations(levelData);
+
+            List<Transform> usedSpawns = SpawnDeliveries(levelData, labels);
+
+            List<Transform> freeSpawns = GetAllFreeSpawns(usedSpawns);
+
+            SpawnFakeDeliveries(freeSpawns, labels);
+        }
+
+        private List<string> GenerateLabelsForStations(StationLevelData[] stationData)
+        {
+            System.Random random = new System.Random();
+            List<string> labels = new();
+
+            for (int i = 0; i < stationData.Length; i++)
+            {
+                TrainStation targetStation = _allStations[i];
+                string label = string.Empty;
+
+                while (label == string.Empty || labels.Contains(label))
+                {
+                    label
+                        = CHARACTERS[random.Next(0, CHARACTERS.Length)].ToString()
+                        + CHARACTERS[random.Next(0, CHARACTERS.Length)].ToString();
+                }
+
+                labels.Add(label);
+                targetStation.UpdateLabelAndID(label, i);
+            }
+
+            return labels;
+        }
+
+        private List<Transform> SpawnDeliveries(StationLevelData[] levelData, List<string> labels)
+        {
+            List<Transform> usedSpawns = new();
+
+            for (int i = levelData.Length - 1; i >= 0; i--)
+            {
+                List<Transform> availableSpawns = GetAvailableSpawnsForStation(i, usedSpawns);
+                int amountToSpawn = levelData[i].minDeliverablePackages;
+
+                while (amountToSpawn > 0 && availableSpawns.Count > 0)
+                {
+                    availableSpawns.Shuffle();
+                    Transform targetSpawn = availableSpawns[0];
+
+                    if (usedSpawns.Contains(targetSpawn))
+                        continue;
+
+                    PackageData package = SpawnBoxDelivery(targetSpawn, labels[i], i);
+
+                    amountToSpawn--;
+                    usedSpawns.Add(targetSpawn);
+                    availableSpawns.Remove(targetSpawn);
+
+                    _totalScore += _succesfullDeliveryBonus + package.GetComponent<BoxDamageable>().Health;
```

Comment:
I think you should call `HandlePackageDelivery()` instead of changing the score here.

Reply:
This isn't changing the score, it is setting the total score. When generating we track how mant deliverable packages are generated so that we can set the total score.

Reply:
Ah I see, small oversight. But for what do we have a `_totalScore` variable?

Reply:
> Ah I see, small oversight. But for what do we have a `_totalScore` variable?

It is to calculate the player's final score, if there is 1 package to deliver and the total score is 15 we can calculate that when they deliver the package and get 10 points they got 2/3rds of the total score, so they would get a rating of 67%. However we want to show the player's their score is yet to be decided but it's a value that will be used later.

Reply:
I changed the name to `TotalAchievableScore`

Reply:
I changed it to `MaxScore` as discussed.